third_party/nanopb/pb.h

```diff
@@ -22,7 +22,7 @@
 /* #define PB_MAX_REQUIRED_FIELDS 256 */
 
 /* Add support for tag numbers > 255 and fields larger than 255 bytes. */
-/* #define PB_FIELD_16BIT 1 */
+#define PB_FIELD_16BIT 1
```

Comment:
We cannot modify this file. The contents of `third_party/` are managed as git submodules.

It shouldn't be needed either. For bazel, we already have it enabled ([relevant change](https://github.com/grpc/grpc/pull/14716/files)). Similarly for `make`-based build systems we already have this defined via compiler args: https://github.com/grpc/grpc/blob/63476cfa5409c6deedffed03a47bbbcb96f17334/templates/Makefile.template#L224

Reply:
We also use Bazel, but our build's dependency on alts_proto still fails with a message saying to define PB_FIELD_16BIT. We're at release 1.11.0, so your linked change didn't seem to do the trick. Ideas?

Reply:
@yihuazhang made that change, he also needed that `#define`. @yihuazhang , any ideas why it may not be working for @deeglaze ?

Reply:
In gRPC, nanopb source is compiled as a one-off target in `grpc_generate_one_off_targets()` in BUILD file, and it's likely that the source has not been re-compiled in @deeglaze's case. How do you use `alts_proto` in your bazel rule? Can I take a look at your BUILD file?

Reply:
It seems `grpc_security_enclave` directly invokes `alts_frame_protector` target in gRPC release without rebuilding the gRPC source. The nanopb flag will be passed in only if gRPC source is re-built. We had the similar problem before in the sense that gRPC directly invokes the nanopb target in its third_party directory, and our strategy was to rebuild the nanopb source in gRPC BUILD file and pass in the flag. 

@apolcyn, Any idea?

Reply:
@dgquintas your statement is a bit wrong - these files are a code drop, and no longer a submodule import. We CAN change these files, and maybe we should?

But more importantly, nanopb is a third party dependency. It shouldn't in fact be a direct dependent of gRPC like what we're currently doing, but rather be a workspace dependent, added in https://github.com/grpc/grpc/blob/master/bazel/grpc_deps.bzl as a [local_repository](https://docs.bazel.build/versions/master/be/workspace.html#local_repository), so that dependents like asylo can bring in their own version.

Reply:
@nicolasnoble ah, yes, I'm living in the past. That being said, when a long time ago we first explored options, I remember there being _reasons_ for not modifying those files. If I'm misremembering or those reasons don't apply any longer, modifying them would indeed be the easiest option. Another consideration is the need to maintain them in sync with the internal version, which comes from the internal third party version, not our own.

Reply:
Yes to that last part, which is why I'm saying it'd make more sense to decouple the library with a local_repository import so that others might swap it with their own.

Reply:
Your message and my edit of my previous one crossed. We are in agreement.

Reply:
I'm not sure how to resolve the package dependency build failure. If Asylo depends on gRPC, but gRPC depends on Asylo (just to get our platform config_setting), then Bazel will balk, no?